source/server/listener_manager_impl.cc

```diff
@@ -490,8 +489,9 @@ void ListenerManagerImpl::onListenerWarmed(ListenerImpl& listener) {
 
   (*existing_warming_listener)->debugLog("warm complete. updating active listener");
   if (existing_active_listener != active_listeners_.end()) {
-    drainListener(std::move(*existing_active_listener));
+    auto listener = std::move(*existing_active_listener);
```

Comment:
What's the reasoning behind this change? Similar question below. Maybe add a small comment about any sequencing requirements?